### ISSUE TITLE
Workaround for switch buffer?

### ISSUE THREAD

**Issue by JordanAnthonyKing** [FrostyX/current-window-only#8]
If you have two windows open, and a buffer was previously opened in one of them, when you switch to that buffer instead of opening in the current window it will be opened in the window it was open in previously and focus will be shifted to that window. Is there a workaround for this?

**Comment by FrostyX** [comment]
Hello @JordanAnthonyKing, thank you for the report. I am not 100% sure that I understand correctly, so let me write a step-by-step how I tried to reproduce this, and please confirm that it is correct, or tell me what I am doing wrong.

1. I vertically split an Emacs frame so that I have two windows
2. I focus the left window and use `M-x switch-to-buffer` to select a buffer named `foo`
3. I focus the right window and use `M-x switch-to-buffer` to select a buffer named `bar`
4. I remain in the right window and use `M-x switch-to-buffer` to select a buffer named `foo` (which is already visible in the left window)

What you expect to happen:
Both windows should show the same `foo` buffer and your focus should remain in the current window
(this exactly happens to me btw)

What I understand, happens for you:
Your focus is changed to the left window, while the right window still shows the `bar` buffer?

**Comment by FrostyX** [comment]
> Apologies, I reenabled this package today and I'm not seeing that behaviour anymore. 

Ok, nice.

> That said, is there a workaround for switch-to-buffer-other-window?

Depends on how it behaves for you now and what do you want it to do. When `current-window-only-mode` is enabled, then doing `M-x switch-to-buffer-other-window` and selecting a buffer should open the buffer in the current window, not the other window. And at least for me, it works like that.

**Comment by JordanAnthonyKing** [comment]
Yes that's how it works for me but I'd like it to retain its old behaviour. Maybe advise those functions to quickly disable and reenable the mode?

**Comment by FrostyX** [comment]
I see. You can try this:

```elisp
(defun switch-to-buffer-other-window-force ()
  (interactive)
  (let ((override #'current-window-only--switch-to-buffer-other-window))
    (advice-remove 'switch-to-buffer-other-window override)
    (call-interactively #'switch-to-buffer-other-window)
    (advice-add 'switch-to-buffer-other-window :override override)))
```

This should work for your use-case. The implementation probably could be easier but I am too big a of an Emacs Lisp noob for that.

Can you please test, so that I can then recommend this workaround in
https://github.com/FrostyX/current-window-only/tree/main?tab=readme-ov-file#tips-and-tricks

**Comment by JordanAnthonyKing** [comment]
Okay, it seems to happen only when that buffer is visible in the other window, does that help narrow it down?

**Comment by FrostyX** [comment]
Not really,
